### ISSUE TITLE
istioctl manifest generate -o does not output CoreDNS manifests

### ISSUE THREAD

**Issue by elfinhe** [istio/istio#21601]
istioctl manifest generate -o does not output CoreDNS manifests to a folder.

I am using istioctl 1.4.4.

istioctl manifest generate -f below.yaml -o /tmp/a-temp-folder/

```
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  coreDNS:
    enabled: true
```
I cannot find coreDNS in the in output folder. By by using `istioctl manifest generate -f below.yaml `, *without `-o`*, I can get coreDNS related manifests.

**Bug description**

**Expected behavior**

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

**Comment by howardjohn** [comment]
isn't it called istiocoredns now @richardwxn

**Comment by richardwxn** [comment]
I think this might be different issue since it is 1.4.4 and it happens only when you are using -o flag?

**Comment by elfinhe** [comment]
> I think this might be different issue since it is 1.4.4 and it happens only when you are using -o flag?

Yes, only with -o. 

istioctl manifest apply or generate > file are working well.

**Comment by esnible** [comment]
Testing with 1.5.0 beta 3 shows below.yaml does not produce coredns:

```
istioctl manifest generate -f /tmp/below.yaml | grep coredns
# AddonComponents istiocoredns component is disabled.
```

`istioctl manifest generate -f /tmp/below.yaml --set addonComponents.istiocoredns.enabled=true -o  /tmp/a-temp-folder/` did generate _/tmp/a-temp-folder/Base/Pilot/AddonComponents/AddonComponents.yaml_ which included the CoreDNS manifests.

**Comment by elfinhe** [comment]
I tested 1.5.0 this morning. It works well. 

This issue (not generating CoreDNS manifests) exists in 1.4.x, with -o flag.

**Comment by howardjohn** [comment]
Closing as this is fixed in 1.5+
